EXAMEN II  de Computacion 2


In [1]:
import pandas as pd
import numpy as np

In [2]:
por = pd.read_excel('data/porcino.xlsx')

In [3]:
mu = pd.read_csv('data/music.csv')

In [5]:
tt = pd.read_csv('data/titanic.csv')

In [7]:
uni = pd.read_csv('data/w_university_r.csv')

Pregunta 1

Sobre el archivo calidad del porcino en chile 'porcino.xlsx'¿Cuantos procinos hembra hay en la region 'Maule' son en total?


In [9]:
por

ID  Espesor Lomo\n  (m.m)  Porcentaje Magro\n CM (%)    Sexo  \
0      NaN                    NaN                        NaN     NaN   
1      1.0                   64.4                       55.8   MACHO   
2      2.0                   64.0                       57.9   MACHO   
3      3.0                   56.4                       58.8   MACHO   
4      4.0                   69.2                       54.6   MACHO   
..     ...                    ...                        ...     ...   
496  496.0                   80.8                       54.5  HEMBRA   
497  497.0                   73.2                       55.6  HEMBRA   
498  498.0                   66.4                       56.1  HEMBRA   
499  499.0                   70.0                       57.4  HEMBRA   
500  500.0                   73.6                       56.5  HEMBRA   

            Region Calidad de la carne  Peso Canal Caliente \n (PCC- Kg)  \
0              NaN                 NaN                               NaN   
1        O´Higgins          Execelente                              85.9   
2        Araucania          Execelente                              80.4   
3    Metropolitana          Execelente                              81.7   
4    Metropolitana            Estandar                              88.6   
..             ...                 ...                               ...   
496      O´Higgins            Estandar                             108.0   
497      O´Higgins          Execelente                              99.5   
498      O´Higgins          Execelente                              88.0   
499      O´Higgins          Execelente                              83.1   
500      O´Higgins          Execelente                             107.8   

     Peso Canal Refrigerada (PCR - Kg)  
0                                  NaN  
1                                 84.3  
2                                 78.5  
3                                 79.6  
4                                 86.8  
..                                 ...  
496                               65.0  
497                               65.0  
498                               68.0  
499                               68.5  
500                               63.8  

[501 rows x 8 columns]

In [17]:
hem = por[por['Sexo']=='HEMBRA']
hem

ID  Espesor Lomo\n  (m.m)  Porcentaje Magro\n CM (%)    Sexo  \
20    20.0                   64.4                       57.7  HEMBRA   
31    31.0                   76.8                       58.0  HEMBRA   
35    35.0                   67.6                       58.7  HEMBRA   
46    46.0                   64.8                       52.3  HEMBRA   
115  115.0                   71.6                       58.9  HEMBRA   
..     ...                    ...                        ...     ...   
496  496.0                   80.8                       54.5  HEMBRA   
497  497.0                   73.2                       55.6  HEMBRA   
498  498.0                   66.4                       56.1  HEMBRA   
499  499.0                   70.0                       57.4  HEMBRA   
500  500.0                   73.6                       56.5  HEMBRA   

        Region Calidad de la carne  Peso Canal Caliente \n (PCC- Kg)  \
20   O´Higgins          Execelente                              82.4   
31   O´Higgins          Execelente                              97.5   
35   O´Higgins          Execelente                              85.8   
46     Bio-Bio            Estandar                              85.5   
115  O´Higgins          Execelente                              79.1   
..         ...                 ...                               ...   
496  O´Higgins            Estandar                             108.0   
497  O´Higgins          Execelente                              99.5   
498  O´Higgins          Execelente                              88.0   
499  O´Higgins          Execelente                              83.1   
500  O´Higgins          Execelente                             107.8   

     Peso Canal Refrigerada (PCR - Kg)  
20                                81.0  
31                                94.0  
35                                84.0  
46                                83.6  
115                               77.4  
..                                 ...  
496                               65.0  
497                               65.0  
498                               68.0  
499                               68.5  
500                               63.8  

[121 rows x 8 columns]

In [24]:
hem_y_re = por[(por['Sexo']=='HEMBRA')&(por['Region']== 'Maule')]
hem_y_re

ID  Espesor Lomo\n  (m.m)  Porcentaje Magro\n CM (%)    Sexo Region  \
128  128.0                   77.6                       57.8  HEMBRA  Maule   
361  361.0                   64.0                       56.6  HEMBRA  Maule   
362  362.0                   57.2                       53.2  HEMBRA  Maule   

    Calidad de la carne  Peso Canal Caliente \n (PCC- Kg)  \
128          Execelente                              94.5   
361          Execelente                              79.3   
362            Estandar                              72.5   

     Peso Canal Refrigerada (PCR - Kg)  
128                               92.6  
361                               77.2  
362                               71.1

**Respuesta = 3 row**

Pregunta 2 

Sobre el Archivo de sobrevivientes del titanic 'titanic.csv'¿Cuantas personas de 38 años sobrevivieron?

In [21]:
tt

Count  PassengerId  Survived  Pclass  \
0        1            1      Died       3   
1        1            2  Survived       1   
2        1            3  Survived       3   
3        1            4  Survived       1   
4        1            5      Died       3   
..     ...          ...       ...     ...   
886      1          887      Died       2   
887      1          888  Survived       1   
888      1          889      Died       3   
889      1          890  Survived       1   
890      1          891      Died       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 13 columns]

In [23]:
sur38= tt[(tt['Survived']=='Survived')&(tt['Age']== 38 )]
sur38

Count  PassengerId  Survived  Pclass  \
1        1            2  Survived       1   
25       1           26  Survived       3   
61       1           62  Survived       1   
224      1          225  Survived       1   
716      1          717  Survived       1   

                                                  Name     Sex   Age  SibSp  \
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
25   Asplund, Mrs. Carl Oscar (Selma Augusta Emilia...  female  38.0      1   
61                                 Icard, Miss. Amelie  female  38.0      0   
224                       Hoyt, Mr. Frederick Maxfield    male  38.0      1   
716                      Endres, Miss. Caroline Louise  female  38.0      0   

     Parch    Ticket      Fare Cabin Embarked  
1        0  PC 17599   71.2833   C85        C  
25       5    347077   31.3875   NaN        S  
61       0    113572   80.0000   B28      NaN  
224      0     19943   90.0000   C93        S  
716      0  PC 17757  227.5250   C45        C

**Respuesta = 5 row**

Pregunta 3

Sobre el Archivo clasificacio Mundial de UNiversidades 'w_university_r.csv'¿Cuantas Universidades de Cyrus aparecen?


In [25]:
uni

World Rank                                        Institution  \
0              1                                 Harvard University   
1              2              Massachusetts Institute of Technology   
2              3                                Stanford University   
3              4                            University of Cambridge   
4              5                               University of Oxford   
...          ...                                                ...   
1995        1996                           Bharathidasan University   
1996        1997                     Federal University of Maranhão   
1997        1998        Jiangxi University of Finance and Economics   
1998        1999                             University of Tarapacá   
1999        2000  National Institute of Mental Health and Neuros...   

            Location  National Rank Education Rank Employability Rank  \
0                USA              1              1                  1   
1                USA              2              4                 12   
2                USA              3             11                  4   
3     United Kingdom              1              3                 25   
4     United Kingdom              2              7                 27   
...              ...            ...            ...                ...   
1995           India             66              -                  -   
1996          Brazil             56              -                  -   
1997           China            302              -                797   
1998           Chile             16              -                  -   
1999           India             67              -                  -   

     Faculty Rank Research Rank  Score  
0               1             1  100.0  
1               2             7   96.7  
2               3             2   95.1  
3               4            10   94.1  
4               9             4   93.3  
...           ...           ...    ...  
1995            -          1921   65.8  
1996            -          1923   65.7  
1997            -          1948   65.7  
1998            -          1925   65.7  
1999            -          1926   65.7  

[2000 rows x 9 columns]

In [27]:
uni.isnull().sum()

World Rank            0
Institution           0
Location              0
National Rank         0
Education Rank        0
Employability Rank    0
Faculty Rank          0
Research Rank         0
Score                 0
dtype: int64

**Respuesta = No hay nulos**

Pregunta 4

Sobre el Archivo Ingresos millones de dolares   Formatos Musicales 'music.csv' ¿En el año 2019, Cual formato tiene el valor mas alto de 'Revenue'?

In [29]:
mu

Year     Format  Revenue [m$]
0    1973        DVD           0.0
1    1973      Other           0.0
2    1973      Tapes        2941.0
3    1973   Download           0.0
4    1973  Streaming           0.0
..    ...        ...           ...
379  2020   Download         653.0
380  2020  Streaming       10075.0
381  2020   Cassette           0.0
382  2020      Vinyl         626.0
383  2020         CD         484.0

[384 rows x 3 columns]

In [31]:
año = mu[mu['Year']== 2019]
año

Year     Format  Revenue [m$]
368  2019        DVD           1.3
369  2019      Other         333.0
370  2019      Tapes           0.0
371  2019   Download         810.0
372  2019  Streaming        8992.0
373  2019   Cassette           0.0
374  2019      Vinyl         492.0
375  2019         CD         639.0

**Respuesta = 372 / 2019 / Streaming / 8992.0**

Pregunta 5

Sobre el Archivo Calidad de Porcino en chile 'procino.xlsx'¿Cuantos porcinos Macho tiene la calidad de carne 'Estandar'?

In [32]:
por

ID  Espesor Lomo\n  (m.m)  Porcentaje Magro\n CM (%)    Sexo  \
0      NaN                    NaN                        NaN     NaN   
1      1.0                   64.4                       55.8   MACHO   
2      2.0                   64.0                       57.9   MACHO   
3      3.0                   56.4                       58.8   MACHO   
4      4.0                   69.2                       54.6   MACHO   
..     ...                    ...                        ...     ...   
496  496.0                   80.8                       54.5  HEMBRA   
497  497.0                   73.2                       55.6  HEMBRA   
498  498.0                   66.4                       56.1  HEMBRA   
499  499.0                   70.0                       57.4  HEMBRA   
500  500.0                   73.6                       56.5  HEMBRA   

            Region Calidad de la carne  Peso Canal Caliente \n (PCC- Kg)  \
0              NaN                 NaN                               NaN   
1        O´Higgins          Execelente                              85.9   
2        Araucania          Execelente                              80.4   
3    Metropolitana          Execelente                              81.7   
4    Metropolitana            Estandar                              88.6   
..             ...                 ...                               ...   
496      O´Higgins            Estandar                             108.0   
497      O´Higgins          Execelente                              99.5   
498      O´Higgins          Execelente                              88.0   
499      O´Higgins          Execelente                              83.1   
500      O´Higgins          Execelente                             107.8   

     Peso Canal Refrigerada (PCR - Kg)  
0                                  NaN  
1                                 84.3  
2                                 78.5  
3                                 79.6  
4                                 86.8  
..                                 ...  
496                               65.0  
497                               65.0  
498                               68.0  
499                               68.5  
500                               63.8  

[501 rows x 8 columns]

In [34]:
mach_estan = por[(por['Sexo']=='MACHO')&(por['Calidad de la carne']== 'Estandar')]
mach_estan

ID  Espesor Lomo\n  (m.m)  Porcentaje Magro\n CM (%)   Sexo  \
4      4.0                   69.2                       54.6  MACHO   
6      6.0                   60.8                       55.0  MACHO   
9      9.0                   58.0                       50.4  MACHO   
12    12.0                   53.2                       54.0  MACHO   
24    24.0                   54.4                       53.9  MACHO   
..     ...                    ...                        ...    ...   
446  446.0                   65.6                       54.6  MACHO   
450  450.0                   62.8                       54.0  MACHO   
474  474.0                   60.0                       54.9  MACHO   
475  475.0                   74.0                       53.1  MACHO   
483  483.0                   66.8                       54.5  MACHO   

            Region Calidad de la carne  Peso Canal Caliente \n (PCC- Kg)  \
4    Metropolitana            Estandar                              88.6   
6    Metropolitana            Estandar                              80.5   
9            Maule            Estandar                              82.3   
12         Bio-Bio            Estandar                              78.2   
24       O´Higgins            Estandar                              74.7   
..             ...                 ...                               ...   
446     Valparaiso            Estandar                              83.9   
450      O´Higgins            Estandar                              90.0   
474  Metropolitana            Estandar                              72.6   
475  Metropolitana            Estandar                              84.6   
483      O´Higgins            Estandar                              88.6   

     Peso Canal Refrigerada (PCR - Kg)  
4                                 86.8  
6                                 78.5  
9                                 77.4  
12                                76.3  
24                                72.2  
..                                 ...  
446                               83.3  
450                               87.2  
474                               72.1  
475                               82.3  
483                               86.8  

[203 rows x 8 columns]

**Respuesta = 203 rows**

Pregunta 6

Sobre el Archivo calidad del Porcino en Chile 'porcino.xlsx'¿Cuantos porcinos Machos de la Region 'O'Higgin' son en total?

In [35]:
Higgin = por[(por['Sexo']=='MACHO')&(por['Region']== 'O´Higgins')]
Higgin

ID  Espesor Lomo\n  (m.m)  Porcentaje Magro\n CM (%)   Sexo  \
1      1.0                   64.4                       55.8  MACHO   
8      8.0                   57.6                       56.5  MACHO   
13    13.0                   60.0                       56.3  MACHO   
15    15.0                   53.6                       58.1  MACHO   
16    16.0                   62.8                       56.7  MACHO   
..     ...                    ...                        ...    ...   
478  478.0                   69.6                       57.0  MACHO   
479  479.0                   71.6                       55.4  MACHO   
482  482.0                   63.2                       55.9  MACHO   
483  483.0                   66.8                       54.5  MACHO   
489  489.0                   66.0                       55.5  MACHO   

        Region Calidad de la carne  Peso Canal Caliente \n (PCC- Kg)  \
1    O´Higgins          Execelente                              85.9   
8    O´Higgins          Execelente                              92.7   
13   O´Higgins          Execelente                              80.6   
15   O´Higgins          Execelente                              83.1   
16   O´Higgins          Execelente                              82.6   
..         ...                 ...                               ...   
478  O´Higgins          Execelente                              84.8   
479  O´Higgins          Execelente                              83.6   
482  O´Higgins          Execelente                              77.5   
483  O´Higgins            Estandar                              88.6   
489  O´Higgins          Execelente                              76.1   

     Peso Canal Refrigerada (PCR - Kg)  
1                                 84.3  
8                                 90.6  
13                                78.7  
15                                80.9  
16                                80.6  
..                                 ...  
478                               84.1  
479                               82.2  
482                               75.3  
483                               86.8  
489                               75.1  

[308 rows x 8 columns]

**Respuesta = 308 rows**

Pregunta 7

Sobre el Archivo Sobrevivientes del Titanic 'titanic.csv' ¿Cuantos valores nulos tiene?

In [37]:
tt.isnull()

Count  PassengerId  Survived  Pclass   Name    Sex    Age  SibSp  Parch  \
0    False        False     False   False  False  False  False  False  False   
1    False        False     False   False  False  False  False  False  False   
2    False        False     False   False  False  False  False  False  False   
3    False        False     False   False  False  False  False  False  False   
4    False        False     False   False  False  False  False  False  False   
..     ...          ...       ...     ...    ...    ...    ...    ...    ...   
886  False        False     False   False  False  False  False  False  False   
887  False        False     False   False  False  False  False  False  False   
888  False        False     False   False  False  False   True  False  False   
889  False        False     False   False  False  False  False  False  False   
890  False        False     False   False  False  False  False  False  False   

     Ticket   Fare  Cabin  Embarked  
0     False  False   True     False  
1     False  False  False     False  
2     False  False   True     False  
3     False  False  False     False  
4     False  False   True     False  
..      ...    ...    ...       ...  
886   False  False   True     False  
887   False  False  False     False  
888   False  False   True     False  
889   False  False  False     False  
890   False  False   True     False  

[891 rows x 13 columns]

In [38]:
tt.isnull().sum()

Count            0
PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

In [39]:
tt.isnull().sum().sum()

866

**Respuesta = 866 nulos en total**

Pregunta 8

Sobre el Archivo clasificacio Mundial de UNiversidades 'w_university_r.csv'¿Cual es el 'Score' de la segunda mejor universidad de Perú en aparecer ?

In [41]:
uni

World Rank                                        Institution  \
0              1                                 Harvard University   
1              2              Massachusetts Institute of Technology   
2              3                                Stanford University   
3              4                            University of Cambridge   
4              5                               University of Oxford   
...          ...                                                ...   
1995        1996                           Bharathidasan University   
1996        1997                     Federal University of Maranhão   
1997        1998        Jiangxi University of Finance and Economics   
1998        1999                             University of Tarapacá   
1999        2000  National Institute of Mental Health and Neuros...   

            Location  National Rank Education Rank Employability Rank  \
0                USA              1              1                  1   
1                USA              2              4                 12   
2                USA              3             11                  4   
3     United Kingdom              1              3                 25   
4     United Kingdom              2              7                 27   
...              ...            ...            ...                ...   
1995           India             66              -                  -   
1996          Brazil             56              -                  -   
1997           China            302              -                797   
1998           Chile             16              -                  -   
1999           India             67              -                  -   

     Faculty Rank Research Rank  Score  
0               1             1  100.0  
1               2             7   96.7  
2               3             2   95.1  
3               4            10   94.1  
4               9             4   93.3  
...           ...           ...    ...  
1995            -          1921   65.8  
1996            -          1923   65.7  
1997            -          1948   65.7  
1998            -          1925   65.7  
1999            -          1926   65.7  

[2000 rows x 9 columns]

In [44]:
per2 = uni[uni['Location']=='Peru']
per2

World Rank                             Institution Location  \
1505        1506    Cayetano Heredia Peruvian University     Peru   
1580        1581  Pontifical Catholic University of Peru     Peru   

      National Rank Education Rank Employability Rank Faculty Rank  \
1505              1              -                  -            -   
1580              2              -               1339            -   

     Research Rank  Score  
1505          1437   67.7  
1580          1506   67.3

**Respuesta = La Pontifical Catholic University of Peru es la segunda mejor unversidad de Perú con 67.3 de score**

Pregunta 9

Sobre el Archivo Sobrevivientes del Titanic 'titanic.csv' ¿Cuantos varones sobrevivieron ?

In [45]:
tt

Count  PassengerId  Survived  Pclass  \
0        1            1      Died       3   
1        1            2  Survived       1   
2        1            3  Survived       3   
3        1            4  Survived       1   
4        1            5      Died       3   
..     ...          ...       ...     ...   
886      1          887      Died       2   
887      1          888  Survived       1   
888      1          889      Died       3   
889      1          890  Survived       1   
890      1          891      Died       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 13 columns]

In [47]:
sobr_var = tt[(tt['Survived']=='Survived')&(tt['Sex']== 'male')]
sobr_var 

Count  PassengerId  Survived  Pclass                             Name  \
17       1           18  Survived       2     Williams, Mr. Charles Eugene   
21       1           22  Survived       2            Beesley, Mr. Lawrence   
23       1           24  Survived       1     Sloper, Mr. William Thompson   
36       1           37  Survived       3                 Mamee, Mr. Hanna   
55       1           56  Survived       1                Woolner, Mr. Hugh   
..     ...          ...       ...     ...                              ...   
838      1          839  Survived       3                  Chip, Mr. Chang   
839      1          840  Survived       1             Marechal, Mr. Pierre   
857      1          858  Survived       1           Daly, Mr. Peter Denis    
869      1          870  Survived       3  Johnson, Master. Harold Theodor   
889      1          890  Survived       1            Behr, Mr. Karl Howell   

      Sex   Age  SibSp  Parch  Ticket     Fare Cabin Embarked  
17   male   NaN      0      0  244373  13.0000   NaN        S  
21   male  34.0      0      0  248698  13.0000   D56        S  
23   male  28.0      0      0  113788  35.5000    A6        S  
36   male   NaN      0      0    2677   7.2292   NaN        C  
55   male   NaN      0      0   19947  35.5000   C52        S  
..    ...   ...    ...    ...     ...      ...   ...      ...  
838  male  32.0      0      0    1601  56.4958   NaN        S  
839  male   NaN      0      0   11774  29.7000   C47        C  
857  male  51.0      0      0  113055  26.5500   E17        S  
869  male   4.0      1      1  347742  11.1333   NaN        S  
889  male  26.0      0      0  111369  30.0000  C148        C  

[109 rows x 13 columns]

**Respuesta = 109 rows**

Pregunta 10

Sobre el Archivo Ingresos millones de dolares Formatos Musicales 'music.csv'¿Cuantos valores nulos tiene?

In [48]:
mu.isnull()

Year  Format  Revenue [m$]
0    False   False         False
1    False   False         False
2    False   False         False
3    False   False         False
4    False   False         False
..     ...     ...           ...
379  False   False         False
380  False   False         False
381  False   False         False
382  False   False         False
383  False   False         False

[384 rows x 3 columns]

In [49]:
mu.isnull().sum()

Year            0
Format          0
Revenue [m$]    0
dtype: int64

**Respuesta = 0 nulos**

Pregunta 11

Sobre el Archivo clasificacio Mundial de UNiversidades 'w_university_r.csv' ¿Cual es el valor de 'Employability Rank' mas alto de universidad de chile en aparecer ?

In [50]:
uni


World Rank                                        Institution  \
0              1                                 Harvard University   
1              2              Massachusetts Institute of Technology   
2              3                                Stanford University   
3              4                            University of Cambridge   
4              5                               University of Oxford   
...          ...                                                ...   
1995        1996                           Bharathidasan University   
1996        1997                     Federal University of Maranhão   
1997        1998        Jiangxi University of Finance and Economics   
1998        1999                             University of Tarapacá   
1999        2000  National Institute of Mental Health and Neuros...   

            Location  National Rank Education Rank Employability Rank  \
0                USA              1              1                  1   
1                USA              2              4                 12   
2                USA              3             11                  4   
3     United Kingdom              1              3                 25   
4     United Kingdom              2              7                 27   
...              ...            ...            ...                ...   
1995           India             66              -                  -   
1996          Brazil             56              -                  -   
1997           China            302              -                797   
1998           Chile             16              -                  -   
1999           India             67              -                  -   

     Faculty Rank Research Rank  Score  
0               1             1  100.0  
1               2             7   96.7  
2               3             2   95.1  
3               4            10   94.1  
4               9             4   93.3  
...           ...           ...    ...  
1995            -          1921   65.8  
1996            -          1923   65.7  
1997            -          1948   65.7  
1998            -          1925   65.7  
1999            -          1926   65.7  

[2000 rows x 9 columns]

In [51]:
chi = uni[uni['Location']=='Chile']
chi

World Rank                                   Institution Location  \
384          385       Pontifical Catholic University of Chile    Chile   
430          431                           University of Chile    Chile   
822          823                      University of Concepción    Chile   
870          871     Federico Santa María Technical University    Chile   
1169        1170                       Andrés Bello University    Chile   
1232        1233                   Austral University of Chile    Chile   
1376        1377                 University of Santiago, Chile    Chile   
1420        1421  Pontifical Catholic University of Valparaíso    Chile   
1448        1449                     University of La Frontera    Chile   
1623        1624                      University of Valparaíso    Chile   
1627        1628              Catholic University of the North    Chile   
1723        1724                     Diego Portales University    Chile   
1817        1818                Autonomous University of Chile    Chile   
1855        1856                           University of Talca    Chile   
1937        1938                      Adolfo Ibáñez University    Chile   
1998        1999                        University of Tarapacá    Chile   

      National Rank Education Rank Employability Rank Faculty Rank  \
384               1            510                207            -   
430               2              -                648            -   
822               3              -                  -            -   
870               4              -               1073            -   
1169              5              -               1629            -   
1232              6              -               1133            -   
1376              7              -                981            -   
1420              8              -                  -            -   
1448              9              -                  -            -   
1623             10              -                  -            -   
1627             11              -                  -            -   
1723             12              -                  -            -   
1817             13              -                  -            -   
1855             14              -                  -            -   
1937             15              -                754            -   
1998             16              -                  -            -   

     Research Rank  Score  
384            389   75.5  
430            412   74.9  
822            786   71.4  
870            829   71.0  
1169          1119   69.3  
1232          1175   68.9  
1376          1321   68.2  
1420          1362   68.0  
1448          1390   67.9  
1623          1551   67.2  
1627          1557   67.1  
1723          1651   66.8  
1817          1748   66.4  
1855          1784   66.3  
1937          1889   66.0  
1998          1925   65.7

**Respuesta = vemos que el mas alto de Employability Rank es 1629 de Andrés Bello University**

Pregunta 12

Sobre el Archivo clasificacio Mundial de UNiversidades 'w_university_r.csv' ¿Cuantas universidades de chile aparecen?

In [52]:
chi = uni[uni['Location']=='Chile']
chi


World Rank                                   Institution Location  \
384          385       Pontifical Catholic University of Chile    Chile   
430          431                           University of Chile    Chile   
822          823                      University of Concepción    Chile   
870          871     Federico Santa María Technical University    Chile   
1169        1170                       Andrés Bello University    Chile   
1232        1233                   Austral University of Chile    Chile   
1376        1377                 University of Santiago, Chile    Chile   
1420        1421  Pontifical Catholic University of Valparaíso    Chile   
1448        1449                     University of La Frontera    Chile   
1623        1624                      University of Valparaíso    Chile   
1627        1628              Catholic University of the North    Chile   
1723        1724                     Diego Portales University    Chile   
1817        1818                Autonomous University of Chile    Chile   
1855        1856                           University of Talca    Chile   
1937        1938                      Adolfo Ibáñez University    Chile   
1998        1999                        University of Tarapacá    Chile   

      National Rank Education Rank Employability Rank Faculty Rank  \
384               1            510                207            -   
430               2              -                648            -   
822               3              -                  -            -   
870               4              -               1073            -   
1169              5              -               1629            -   
1232              6              -               1133            -   
1376              7              -                981            -   
1420              8              -                  -            -   
1448              9              -                  -            -   
1623             10              -                  -            -   
1627             11              -                  -            -   
1723             12              -                  -            -   
1817             13              -                  -            -   
1855             14              -                  -            -   
1937             15              -                754            -   
1998             16              -                  -            -   

     Research Rank  Score  
384            389   75.5  
430            412   74.9  
822            786   71.4  
870            829   71.0  
1169          1119   69.3  
1232          1175   68.9  
1376          1321   68.2  
1420          1362   68.0  
1448          1390   67.9  
1623          1551   67.2  
1627          1557   67.1  
1723          1651   66.8  
1817          1748   66.4  
1855          1784   66.3  
1937          1889   66.0  
1998          1925   65.7

**Respuesta = Aparecen 16 universidades**

Pregunta 13

Sobre el Archivo Calidad de Porcino en chile 'procino.xlsx' ¿Cuantos porcinos tiene la calidad de carne 'Estandar'?

In [57]:
por

ID  Espesor Lomo\n  (m.m)  Porcentaje Magro\n CM (%)    Sexo  \
0      NaN                    NaN                        NaN     NaN   
1      1.0                   64.4                       55.8   MACHO   
2      2.0                   64.0                       57.9   MACHO   
3      3.0                   56.4                       58.8   MACHO   
4      4.0                   69.2                       54.6   MACHO   
..     ...                    ...                        ...     ...   
496  496.0                   80.8                       54.5  HEMBRA   
497  497.0                   73.2                       55.6  HEMBRA   
498  498.0                   66.4                       56.1  HEMBRA   
499  499.0                   70.0                       57.4  HEMBRA   
500  500.0                   73.6                       56.5  HEMBRA   

            Region Calidad de la carne  Peso Canal Caliente \n (PCC- Kg)  \
0              NaN                 NaN                               NaN   
1        O´Higgins          Execelente                              85.9   
2        Araucania          Execelente                              80.4   
3    Metropolitana          Execelente                              81.7   
4    Metropolitana            Estandar                              88.6   
..             ...                 ...                               ...   
496      O´Higgins            Estandar                             108.0   
497      O´Higgins          Execelente                              99.5   
498      O´Higgins          Execelente                              88.0   
499      O´Higgins          Execelente                              83.1   
500      O´Higgins          Execelente                             107.8   

     Peso Canal Refrigerada (PCR - Kg)  
0                                  NaN  
1                                 84.3  
2                                 78.5  
3                                 79.6  
4                                 86.8  
..                                 ...  
496                               65.0  
497                               65.0  
498                               68.0  
499                               68.5  
500                               63.8  

[501 rows x 8 columns]

In [59]:
est = por[por['Calidad de la carne']=='Estandar']
est 

ID  Espesor Lomo\n  (m.m)  Porcentaje Magro\n CM (%)    Sexo  \
4      4.0                   69.2                       54.6   MACHO   
6      6.0                   60.8                       55.0   MACHO   
9      9.0                   58.0                       50.4   MACHO   
12    12.0                   53.2                       54.0   MACHO   
24    24.0                   54.4                       53.9   MACHO   
..     ...                    ...                        ...     ...   
483  483.0                   66.8                       54.5   MACHO   
486  486.0                   68.8                       55.0  HEMBRA   
490  490.0                   72.4                       55.0  HEMBRA   
493  493.0                   73.2                       52.9  HEMBRA   
496  496.0                   80.8                       54.5  HEMBRA   

            Region Calidad de la carne  Peso Canal Caliente \n (PCC- Kg)  \
4    Metropolitana            Estandar                              88.6   
6    Metropolitana            Estandar                              80.5   
9            Maule            Estandar                              82.3   
12         Bio-Bio            Estandar                              78.2   
24       O´Higgins            Estandar                              74.7   
..             ...                 ...                               ...   
483      O´Higgins            Estandar                              88.6   
486  Metropolitana            Estandar                              82.0   
490      O´Higgins            Estandar                              93.7   
493      O´Higgins            Estandar                             108.5   
496      O´Higgins            Estandar                             108.0   

     Peso Canal Refrigerada (PCR - Kg)  
4                                 86.8  
6                                 78.5  
9                                 77.4  
12                                76.3  
24                                72.2  
..                                 ...  
483                               86.8  
486                               80.6  
490                               69.1  
493                               66.1  
496                               65.0  

[230 rows x 8 columns]

**Respuesta = 230 rows**

Pregunta 14

Sobre el Archivo clasificacio Mundial de UNiversidades 'w_university_r.csv' ¿Cual es el 'Score' de la mejor Universidad de Chile en aparecer?

In [60]:
uni


World Rank                                        Institution  \
0              1                                 Harvard University   
1              2              Massachusetts Institute of Technology   
2              3                                Stanford University   
3              4                            University of Cambridge   
4              5                               University of Oxford   
...          ...                                                ...   
1995        1996                           Bharathidasan University   
1996        1997                     Federal University of Maranhão   
1997        1998        Jiangxi University of Finance and Economics   
1998        1999                             University of Tarapacá   
1999        2000  National Institute of Mental Health and Neuros...   

            Location  National Rank Education Rank Employability Rank  \
0                USA              1              1                  1   
1                USA              2              4                 12   
2                USA              3             11                  4   
3     United Kingdom              1              3                 25   
4     United Kingdom              2              7                 27   
...              ...            ...            ...                ...   
1995           India             66              -                  -   
1996          Brazil             56              -                  -   
1997           China            302              -                797   
1998           Chile             16              -                  -   
1999           India             67              -                  -   

     Faculty Rank Research Rank  Score  
0               1             1  100.0  
1               2             7   96.7  
2               3             2   95.1  
3               4            10   94.1  
4               9             4   93.3  
...           ...           ...    ...  
1995            -          1921   65.8  
1996            -          1923   65.7  
1997            -          1948   65.7  
1998            -          1925   65.7  
1999            -          1926   65.7  

[2000 rows x 9 columns]

In [62]:
chi_may= uni[uni['Location']=='Chile']
chi_may

World Rank                                   Institution Location  \
384          385       Pontifical Catholic University of Chile    Chile   
430          431                           University of Chile    Chile   
822          823                      University of Concepción    Chile   
870          871     Federico Santa María Technical University    Chile   
1169        1170                       Andrés Bello University    Chile   
1232        1233                   Austral University of Chile    Chile   
1376        1377                 University of Santiago, Chile    Chile   
1420        1421  Pontifical Catholic University of Valparaíso    Chile   
1448        1449                     University of La Frontera    Chile   
1623        1624                      University of Valparaíso    Chile   
1627        1628              Catholic University of the North    Chile   
1723        1724                     Diego Portales University    Chile   
1817        1818                Autonomous University of Chile    Chile   
1855        1856                           University of Talca    Chile   
1937        1938                      Adolfo Ibáñez University    Chile   
1998        1999                        University of Tarapacá    Chile   

      National Rank Education Rank Employability Rank Faculty Rank  \
384               1            510                207            -   
430               2              -                648            -   
822               3              -                  -            -   
870               4              -               1073            -   
1169              5              -               1629            -   
1232              6              -               1133            -   
1376              7              -                981            -   
1420              8              -                  -            -   
1448              9              -                  -            -   
1623             10              -                  -            -   
1627             11              -                  -            -   
1723             12              -                  -            -   
1817             13              -                  -            -   
1855             14              -                  -            -   
1937             15              -                754            -   
1998             16              -                  -            -   

     Research Rank  Score  
384            389   75.5  
430            412   74.9  
822            786   71.4  
870            829   71.0  
1169          1119   69.3  
1232          1175   68.9  
1376          1321   68.2  
1420          1362   68.0  
1448          1390   67.9  
1623          1551   67.2  
1627          1557   67.1  
1723          1651   66.8  
1817          1748   66.4  
1855          1784   66.3  
1937          1889   66.0  
1998          1925   65.7

**Respuesta = Es 75.5 de Pontifical Catholic University of Chile**

Pregunta 15

Sobre el Archivo Calidad de Porcino en chile 'procino.xls' ¿Cuantos porcinos hembra de la region 'Metropolitana'son en total?

In [63]:
por

ID  Espesor Lomo\n  (m.m)  Porcentaje Magro\n CM (%)    Sexo  \
0      NaN                    NaN                        NaN     NaN   
1      1.0                   64.4                       55.8   MACHO   
2      2.0                   64.0                       57.9   MACHO   
3      3.0                   56.4                       58.8   MACHO   
4      4.0                   69.2                       54.6   MACHO   
..     ...                    ...                        ...     ...   
496  496.0                   80.8                       54.5  HEMBRA   
497  497.0                   73.2                       55.6  HEMBRA   
498  498.0                   66.4                       56.1  HEMBRA   
499  499.0                   70.0                       57.4  HEMBRA   
500  500.0                   73.6                       56.5  HEMBRA   

            Region Calidad de la carne  Peso Canal Caliente \n (PCC- Kg)  \
0              NaN                 NaN                               NaN   
1        O´Higgins          Execelente                              85.9   
2        Araucania          Execelente                              80.4   
3    Metropolitana          Execelente                              81.7   
4    Metropolitana            Estandar                              88.6   
..             ...                 ...                               ...   
496      O´Higgins            Estandar                             108.0   
497      O´Higgins          Execelente                              99.5   
498      O´Higgins          Execelente                              88.0   
499      O´Higgins          Execelente                              83.1   
500      O´Higgins          Execelente                             107.8   

     Peso Canal Refrigerada (PCR - Kg)  
0                                  NaN  
1                                 84.3  
2                                 78.5  
3                                 79.6  
4                                 86.8  
..                                 ...  
496                               65.0  
497                               65.0  
498                               68.0  
499                               68.5  
500                               63.8  

[501 rows x 8 columns]

In [64]:
metropo  = por[(por['Sexo']=='HEMBRA')&(por['Region']== 'Metropolitana')]
metropo

ID  Espesor Lomo\n  (m.m)  Porcentaje Magro\n CM (%)    Sexo  \
485  485.0                   70.8                       57.5  HEMBRA   
486  486.0                   68.8                       55.0  HEMBRA   
487  487.0                   74.8                       58.8  HEMBRA   
488  488.0                   57.6                       55.7  HEMBRA   

            Region Calidad de la carne  Peso Canal Caliente \n (PCC- Kg)  \
485  Metropolitana          Execelente                              92.1   
486  Metropolitana            Estandar                              82.0   
487  Metropolitana          Execelente                              83.5   
488  Metropolitana          Execelente                              89.0   

     Peso Canal Refrigerada (PCR - Kg)  
485                               90.2  
486                               80.6  
487                               81.3  
488                               64.9

**Respuesta = son en total 4**

Pregunta 16

Sobre el Archivo Sobrevivientes del Titanic 'titanic.csv' ¿Cuantos varones murieron?

In [66]:
tt


Count  PassengerId  Survived  Pclass  \
0        1            1      Died       3   
1        1            2  Survived       1   
2        1            3  Survived       3   
3        1            4  Survived       1   
4        1            5      Died       3   
..     ...          ...       ...     ...   
886      1          887      Died       2   
887      1          888  Survived       1   
888      1          889      Died       3   
889      1          890  Survived       1   
890      1          891      Died       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 13 columns]

In [67]:
var_muertos = tt[(tt['Sex']== 'male')&(tt['Survived']=='Died')]
var_muertos

Count  PassengerId Survived  Pclass                            Name  \
0        1            1     Died       3         Braund, Mr. Owen Harris   
4        1            5     Died       3        Allen, Mr. William Henry   
5        1            6     Died       3                Moran, Mr. James   
6        1            7     Died       1         McCarthy, Mr. Timothy J   
7        1            8     Died       3  Palsson, Master. Gosta Leonard   
..     ...          ...      ...     ...                             ...   
881      1          882     Died       3              Markun, Mr. Johann   
883      1          884     Died       2   Banfield, Mr. Frederick James   
884      1          885     Died       3          Sutehall, Mr. Henry Jr   
886      1          887     Died       2           Montvila, Rev. Juozas   
890      1          891     Died       3             Dooley, Mr. Patrick   

      Sex   Age  SibSp  Parch            Ticket     Fare Cabin Embarked  
0    male  22.0      1      0         A/5 21171   7.2500   NaN        S  
4    male  35.0      0      0            373450   8.0500   NaN        S  
5    male   NaN      0      0            330877   8.4583   NaN        Q  
6    male  54.0      0      0             17463  51.8625   E46        S  
7    male   2.0      3      1            349909  21.0750   NaN        S  
..    ...   ...    ...    ...               ...      ...   ...      ...  
881  male  33.0      0      0            349257   7.8958   NaN        S  
883  male  28.0      0      0  C.A./SOTON 34068  10.5000   NaN        S  
884  male  25.0      0      0   SOTON/OQ 392076   7.0500   NaN        S  
886  male  27.0      0      0            211536  13.0000   NaN        S  
890  male  32.0      0      0            370376   7.7500   NaN        Q  

[468 rows x 13 columns]

**Respuesta = 468 rows**

Pregunta 17

Sobre el Archivo Sobrevivientes del Titanic 'titanic.csv' ¿Cuantas mujeres de 38 sobrevivieron?


In [70]:
tt

Count  PassengerId  Survived  Pclass  \
0        1            1      Died       3   
1        1            2  Survived       1   
2        1            3  Survived       3   
3        1            4  Survived       1   
4        1            5      Died       3   
..     ...          ...       ...     ...   
886      1          887      Died       2   
887      1          888  Survived       1   
888      1          889      Died       3   
889      1          890  Survived       1   
890      1          891      Died       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 13 columns]

In [71]:
mu_sur38= tt[(tt['Survived']=='Survived')&(tt['Age']== 38)&(tt['Sex']=='female')]
mu_sur38

Count  PassengerId  Survived  Pclass  \
1        1            2  Survived       1   
25       1           26  Survived       3   
61       1           62  Survived       1   
716      1          717  Survived       1   

                                                  Name     Sex   Age  SibSp  \
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
25   Asplund, Mrs. Carl Oscar (Selma Augusta Emilia...  female  38.0      1   
61                                 Icard, Miss. Amelie  female  38.0      0   
716                      Endres, Miss. Caroline Louise  female  38.0      0   

     Parch    Ticket      Fare Cabin Embarked  
1        0  PC 17599   71.2833   C85        C  
25       5    347077   31.3875   NaN        S  
61       0    113572   80.0000   B28      NaN  
716      0  PC 17757  227.5250   C45        C

**Respuesta = solo 4**

Pregunta 18

Sobre el Archivo Calidad de Porcino en chile 'procino.xls' ¿Cuantos porcinos hemnra de la region 'O'Higgin' son en total?

In [72]:
por

ID  Espesor Lomo\n  (m.m)  Porcentaje Magro\n CM (%)    Sexo  \
0      NaN                    NaN                        NaN     NaN   
1      1.0                   64.4                       55.8   MACHO   
2      2.0                   64.0                       57.9   MACHO   
3      3.0                   56.4                       58.8   MACHO   
4      4.0                   69.2                       54.6   MACHO   
..     ...                    ...                        ...     ...   
496  496.0                   80.8                       54.5  HEMBRA   
497  497.0                   73.2                       55.6  HEMBRA   
498  498.0                   66.4                       56.1  HEMBRA   
499  499.0                   70.0                       57.4  HEMBRA   
500  500.0                   73.6                       56.5  HEMBRA   

            Region Calidad de la carne  Peso Canal Caliente \n (PCC- Kg)  \
0              NaN                 NaN                               NaN   
1        O´Higgins          Execelente                              85.9   
2        Araucania          Execelente                              80.4   
3    Metropolitana          Execelente                              81.7   
4    Metropolitana            Estandar                              88.6   
..             ...                 ...                               ...   
496      O´Higgins            Estandar                             108.0   
497      O´Higgins          Execelente                              99.5   
498      O´Higgins          Execelente                              88.0   
499      O´Higgins          Execelente                              83.1   
500      O´Higgins          Execelente                             107.8   

     Peso Canal Refrigerada (PCR - Kg)  
0                                  NaN  
1                                 84.3  
2                                 78.5  
3                                 79.6  
4                                 86.8  
..                                 ...  
496                               65.0  
497                               65.0  
498                               68.0  
499                               68.5  
500                               63.8  

[501 rows x 8 columns]

In [73]:
he_higg = por[(por['Sexo']=='HEMBRA')&(por['Region']== 'O´Higgins')]
he_higg

ID  Espesor Lomo\n  (m.m)  Porcentaje Magro\n CM (%)    Sexo  \
20    20.0                   64.4                       57.7  HEMBRA   
31    31.0                   76.8                       58.0  HEMBRA   
35    35.0                   67.6                       58.7  HEMBRA   
115  115.0                   71.6                       58.9  HEMBRA   
116  116.0                   75.6                       60.0  HEMBRA   
..     ...                    ...                        ...     ...   
496  496.0                   80.8                       54.5  HEMBRA   
497  497.0                   73.2                       55.6  HEMBRA   
498  498.0                   66.4                       56.1  HEMBRA   
499  499.0                   70.0                       57.4  HEMBRA   
500  500.0                   73.6                       56.5  HEMBRA   

        Region Calidad de la carne  Peso Canal Caliente \n (PCC- Kg)  \
20   O´Higgins          Execelente                              82.4   
31   O´Higgins          Execelente                              97.5   
35   O´Higgins          Execelente                              85.8   
115  O´Higgins          Execelente                              79.1   
116  O´Higgins          Execelente                              74.6   
..         ...                 ...                               ...   
496  O´Higgins            Estandar                             108.0   
497  O´Higgins          Execelente                              99.5   
498  O´Higgins          Execelente                              88.0   
499  O´Higgins          Execelente                              83.1   
500  O´Higgins          Execelente                             107.8   

     Peso Canal Refrigerada (PCR - Kg)  
20                                81.0  
31                                94.0  
35                                84.0  
115                               77.4  
116                               73.1  
..                                 ...  
496                               65.0  
497                               65.0  
498                               68.0  
499                               68.5  
500                               63.8  

[106 rows x 8 columns]

**Respuesta = 106 rows**

Pregunta 19

Sobre el Archivo Ingresos millones de dolares Formatos Musicales 'music.csv' ¿El 'Format' = Streaming en que año obtuvo el valor mas alto en Revenue[m]?

In [74]:
mu

Year     Format  Revenue [m$]
0    1973        DVD           0.0
1    1973      Other           0.0
2    1973      Tapes        2941.0
3    1973   Download           0.0
4    1973  Streaming           0.0
..    ...        ...           ...
379  2020   Download         653.0
380  2020  Streaming       10075.0
381  2020   Cassette           0.0
382  2020      Vinyl         626.0
383  2020         CD         484.0

[384 rows x 3 columns]

In [75]:
stream = mu[(mu['Format']=='Streaming')]
stream

Year     Format  Revenue [m$]
4    1973  Streaming           0.0
12   1974  Streaming           0.0
20   1975  Streaming           0.0
28   1976  Streaming           0.0
36   1977  Streaming           0.0
44   1978  Streaming           0.0
52   1979  Streaming           0.0
60   1980  Streaming           0.0
68   1981  Streaming           0.0
76   1982  Streaming           0.0
84   1983  Streaming           0.0
92   1984  Streaming           0.0
100  1985  Streaming           0.0
108  1986  Streaming           0.0
116  1987  Streaming           0.0
124  1988  Streaming           0.0
132  1989  Streaming           0.0
140  1990  Streaming           0.0
148  1991  Streaming           0.0
156  1992  Streaming           0.0
164  1993  Streaming           0.0
172  1994  Streaming           0.0
180  1995  Streaming           0.0
188  1996  Streaming           0.0
196  1997  Streaming           0.0
204  1998  Streaming           0.0
212  1999  Streaming           0.0
220  2000  Streaming           0.0
228  2001  Streaming           0.0
236  2002  Streaming           0.0
244  2003  Streaming           0.0
252  2004  Streaming           9.5
260  2005  Streaming         225.0
268  2006  Streaming         307.0
276  2007  Streaming         337.0
284  2008  Streaming         386.0
292  2009  Streaming         436.0
300  2010  Streaming         548.0
308  2011  Streaming         752.0
316  2012  Streaming        1164.0
324  2013  Streaming        1616.0
332  2014  Streaming        1998.0
340  2015  Streaming        2546.0
348  2016  Streaming        4179.0
356  2017  Streaming        5721.0
364  2018  Streaming        7491.0
372  2019  Streaming        8992.0
380  2020  Streaming       10075.0

**Respuesta = El año con mas Revenue [m$] es el 2020**

Pregunta 20

Sobre el Archivo Ingresos millones de dolares Formatos Musicales 'music.csv' ¿El 'Format' = Vinyl en que año obtuvo el valor mas alto en Revenue[m]?


In [77]:
stream_v = mu[(mu['Format']=='Vinyl')]
stream_v

Year Format  Revenue [m$]
6    1973  Vinyl        8371.0
14   1974  Vinyl        8137.0
22   1975  Vinyl        8161.0
30   1976  Vinyl        8679.0
38   1977  Vinyl       10422.0
46   1978  Vinyl       10851.0
54   1979  Vinyl        8594.0
62   1980  Vinyl        7695.0
70   1981  Vinyl        7397.0
78   1982  Vinyl        5922.0
86   1983  Vinyl        5089.0
94   1984  Vinyl        4602.0
102  1985  Vinyl        3756.0
110  1986  Vinyl        2860.0
118  1987  Vinyl        2270.0
126  1988  Vinyl        1559.0
134  1989  Vinyl         703.0
142  1990  Vinyl         358.0
150  1991  Vinyl         177.0
158  1992  Vinyl         147.0
166  1993  Vinyl         111.0
174  1994  Vinyl         114.0
182  1995  Vinyl         122.0
190  1996  Vinyl         139.0
198  1997  Vinyl         111.0
206  1998  Vinyl          95.0
214  1999  Vinyl          93.0
222  2000  Vinyl          81.0
230  2001  Vinyl          86.0
238  2002  Vinyl          65.0
246  2003  Vinyl          61.0
254  2004  Vinyl          54.0
262  2005  Vinyl          36.0
270  2006  Vinyl          33.0
278  2007  Vinyl          34.0
286  2008  Vinyl          72.0
294  2009  Vinyl          80.0
302  2010  Vinyl         108.0
310  2011  Vinyl         143.0
318  2012  Vinyl         187.0
326  2013  Vinyl         237.0
334  2014  Vinyl         273.0
342  2015  Vinyl         370.0
350  2016  Vinyl         389.0
358  2017  Vinyl         417.0
366  2018  Vinyl         438.0
374  2019  Vinyl         492.0
382  2020  Vinyl         626.0

**Respuesta = El mayor es 1978 con 10851.0**